iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -209,6 +210,197 @@ class GlobalStoreOpConversion : public OpConversionPattern<StoreOpTy> {
   StringRef funcName;
 };
 
+class ListAllocOpConversion
+    : public OpConversionPattern<IREE::VM::ListAllocOp> {
+  using OpConversionPattern<IREE::VM::ListAllocOp>::OpConversionPattern;
+
+ private:
+  LogicalResult matchAndRewrite(
+      IREE::VM::ListAllocOp allocOp, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    // clang-format off
+    // The generated c code looks roughly like this.
+    // iree_vm_type_def_t element_type = iree_vm_type_def_make_value_type(IREE_VM_VALUE_TYPE_I32);
+    // iree_vm_type_def_t* element_type_ptr = &element_type;
+    // iree_vm_list_t* list = nullptr;
```

Comment:
`nullptr` or `NULL`? :confused:

Reply:
If the generated code is pure C, and I think it should be, `NULL` should be chosen. `nullptr` is the null pointer literal in C++.